refarch-integrations/refarch-s3-integration/refarch-s3-integration-core/src/main/java/de/muenchen/refarch/integration/s3/adapter/out/s3/S3Adapter.java

```diff
@@ -121,6 +124,43 @@ public Set<String> getFilePathsFromFolder(final String folder) throws FileSystem
         }
     }
 
+    /**
+     * Returns the metadata of the files in a given folder.
+     *
+     * @param folder The folder. The path must be absolute and without specifying the bucket. Example 1:
+     *            Folder in bucket: "BUCKET/folder" Specification in
+     *            parameter: "folder" Example 2: Folder in bucket: "BUCKET/folder/subfolder"
+     *            Specification in parameter: "folder/subfolder"
+     * @return the metadata of the files in a given folder. Also returns the metadata of the files in
+     *         subfolders.
+     * @throws FileSystemAccessException if the paths cannot be downloaded.
+     */
+    @Override
+    public List<FileMetadata> getMetadataOfFilesFromFolder(final String folder) throws FileSystemAccessException {
+        try {
+            final ListObjectsArgs listObjectsArgs = ListObjectsArgs.builder()
+                    .bucket(this.bucketName)
+                    .prefix(folder)
+                    .recursive(true)
+                    .build();
+            final List<Result<Item>> resultItemList = IteratorUtils.toList(this.client.listObjects(listObjectsArgs).iterator());
+            final List<FileMetadata> fileInformationsFromFolder = new ArrayList<>();
+            for (final Result<Item> resultItem : resultItemList) {
+                final FileMetadata fileInformation = new FileMetadata(
+                        resultItem.get().objectName(),
+                        resultItem.get().size(),
+                        resultItem.get().etag(),
```

Comment:
Should we probably remove the duplicate quotation marks?

![grafik](https://github.com/user-attachments/assets/6b264f19-92d9-4ac5-9012-f94f68e97a1a)

Reply:
yeah, that makes sense. I will implement a removal logic.

Reply:
quotation marks removal implementet.

https://github.com/it-at-m/refarch/pull/473/commits/a5885c69b60cfe9f93e496323150e52982fbed08